hcloud/_client.py

```diff
@@ -231,3 +233,44 @@ def request(  # type: ignore[no-untyped-def]
 
         # TODO: return an empty dict instead of an empty string when content == "".
         return content  # type: ignore[return-value]
+
+    def session(self, session: requests.Session) -> None:
+        """
+        Configure a custom :class:`Session <requests.Session>` to use when calling the API.
+
+        :param session: The session to use when making API requests.
+        """
+        self._requests_session = session
+
+    @contextmanager
+    def cached_session(self) -> Generator[Client, None, None]:
+        """
+        Provide a copy of the :class:`Client <hcloud.Client>` as context manager that
+        will cache all GET requests.
+
+        Cached response will not expire automatically, therefore the cached client must
+        not be used for long living scopes.
+        """
+        client = copy.deepcopy(self)
+        client.session(CachedSession())
+        yield client
```

Comment:
I am not yet convinced about this implementation; I might only allow users to configure the session, and move the  CachedSession code to the ansible inventory project for the time we need to iron it out.